cpp/ql/lib/semmle/code/cpp/models/implementations/Pure.qll

```diff
@@ -26,22 +27,43 @@ private class PureStrFunction extends AliasFunction, ArrayFunction, TaintFunctio
   }
 
   override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
-    exists(ParameterIndex i |
+    exists(ParameterIndex i | exists(this.getParameter(i)) |
+      // For these functions we add taint flow according to the following rules:
+      // 1. If the parameter is of a pointer type then there is taint from the
+      // indirection of the parameter. Otherwise, there is taint from the
+      // parameter.
+      // 2. If the return value is of a pointer type then there is taint to the
+      // indirection of the return. Otherwise, there is taint to the return.
```

Comment:
Does it make sense to move this comment up before the `exists` given that it concerns both the stuff inside the `exists` and also the conjunct after it?

Reply:
Sure. Fixed in 67e3b6999650b66b131ba3fd8738d3b839ebd2bf